### DISCUSSION TITLE
Add Telemetry sensors for helicopter use

### DISCUSSION THREAD

**Discussion by ahmad-kassem** [ExpressLRS/ExpressLRS#1858] [Ideas]
Hi

it would be very useful if you can add some additional telemetry sensors to the list of available sensors with OpenTX.
These sensors will be used by the RotorFlight Dev team to send additional telemetry to the radio that are specific and requested among Helicopters pilots.

these sensors are:
1- HeadSpeed [RPM]
2- Battery Pack Voltage [V]
3-ESC temperature [°C]
4- Vibration Level [%]
5- Throttle Input [%]
6- ESC Power Out [%]

**Comment by ZZ-Cat** [comment]
I was going to suggest precisely this, & you bet me to it.
I am about to start converting all of my helicopters over to RotorFlight & ExpressLRS, starting with my Goblin RAW 420.

It would be great for us heli pilots to have this, as it will complement the 8-channel full resolution feature quite nicely. =^/,..,^=

**Comment by schugabe** [comment]
Hi!

Battery Voltage is already supported.

I get why head speed and temperature would be interessting.

But what would a vibraton level be?
The TX is already sending the throttle input? Why would that be needed as telemetry value? Same with ESC power out?

**Reply by ZZ-Cat** [reply]
IIRC, vibration is a metric measured by the flight controller itself to measure vibration levels in the helicopter (usually for diagnosing mechanical issues in flight).
As far as I know, not many flight controllers have vibration as a telemetry value, as this is often self-contained & is instead recorded into memory within the flight controller itself.
This is certainly the case with the MSH Brain2, & RotorFlight flight controllers.
So, you can't really use something that isn't there in the first place.

ESC power output could be used to more - accurately calculate things like energy consumption (EG mAh counter).
But, again, the ESC itself would need to provide this as telemetry output to the receiver.

**Comment by JyeSmith** [comment]
We are asking TBS if they are interested in creating a crsf repo so that it can be expanded on in a controlled way e.g. new sensors etc

Please feel free to comment or add your support if you feel inclined 🙂

https://github.com/tbs-fpv/freedomtx/issues/26

**Reply by ZZ-Cat** [reply]
Awesome. Will do.

**Comment by sp3swj** [comment]
Is any plan to add Temperature Telemetry - for example just some Tmp1 Tmp2... Tmp8  to send general temperature by CRSF>ELRS to LCD transmitter.  Currently i temporary make some "**workaround**"  sending temperature  using "battery fiels" and manually modify properties  - but this is not PRO solution :-)

Or just add some "general telemetry" field for any simple use - to send something.

![image](https://github.com/user-attachments/assets/ddace483-7986-4412-9729-338eb53d0ac3)  

![image](https://github.com/user-attachments/assets/cd074c23-1be8-41e7-98d5-e04b2d0a8a39)


I saw some discussion about temperature  here  https://github.com/rotorflight/rotorflight-firmware/issues/81#issuecomment-2160375732

maybe good idea is to add some new telemetry CRSF  to   EdgeTX   :-)

**Reply by moadmoosa** [reply]
Can you share how you did it?  I want to make something similar like this.

Thanks

**Comment by bigtwin** [comment]
I am also interested in a temperatur sensor. I would like to measure 3 or 4 different temperature and and have them transmitted via ELRS. Is there any sensor available to the public which I can more or less just buy and connect? 
I am able to do some minor electronic work and soldering but not any programming. 
Thx for your information!
